libfreerdp/core/info.c

```diff
@@ -782,13 +775,16 @@ static void rdp_write_info_packet(rdpRdp* rdp, wStream* s)
 		cbUserName = ConvertToUnicode(CP_UTF8, 0, settings->Username, -1, &userNameW, 0) * 2;
 	}
 
+	/* excludes (!) the length of the mandatory null terminator */
+	cbUserName = cbUserName > 0 ? cbUserName - 2 : 0;
```

Comment:
I´d change that to `cbUserName = cbUserName > 2 ? cbUserName - 2 : cbUserName;` to avoid a possible overflow